cpp/src/arrow/dataset/scanner_test.cc

```diff
@@ -1074,6 +1074,24 @@ TEST_P(TestScanner, ProjectedScanNested) {
   AssertScanBatchesUnorderedEqualRepetitionsOf(MakeScanner(batch_in), batch_out);
 }
 
+TEST_P(TestScanner, ProjectedScanNestedFromNames) {
+  SetSchema({
+      field("struct", struct_({field("i32", int32()), field("f64", float64())})),
+      field("nested", struct_({field("left", int32()),
+                               field("right", struct_({field("i32", int32()),
+                                                       field("f64", float64())}))})),
+  });
+  ASSERT_OK_AND_ASSIGN(auto descr,
+                       ProjectionDescr::FromNames({"struct.i32", "nested.right.f64"},
```

Comment:
```suggestion
                       ProjectionDescr::FromNames({".struct.i32", "nested.right.f64"},
```

(so we cover both with and without leading dot)